agent/logging.go

```diff
@@ -74,36 +77,46 @@ func GetSender(ctx context.Context, prefix, taskId string) (send.Sender, error)
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
-func (a *Agent) makeLoggerProducer(ctx context.Context, c *model.LoggerConfig, td client.TaskData) client.LoggerProducer {
-	config := convertLoggerConfig(*c, a.opts.WorkingDirectory)
+func (a *Agent) makeLoggerProducer(ctx context.Context, c *model.LoggerConfig, td client.TaskData, task *task.Task) client.LoggerProducer {
+	path := fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, taskLogDirectory)
+	grip.Error(errors.Wrap(os.Mkdir(path, os.ModeDir|os.ModePerm), "error making log directory"))
+	config := convertLoggerConfig(*c, path, a.opts.LogkeeperURL, td.ID, task.Execution)
 	return a.comm.GetLoggerProducer(ctx, td, &config)
 }
 
-// TODO: configuration for logkeeper
-func convertLoggerConfig(c model.LoggerConfig, workdir string) client.LoggerConfig {
+func convertLoggerConfig(c model.LoggerConfig, workdir, logkeeperURL, taskID string, execution int) client.LoggerConfig {
```

Comment:
should all these args be in some sort of POD struct?

Reply:
made it part of the function above so we don't need to do that